apisx/v1alpha1/xlistenerset_types.go

```diff
@@ -152,7 +152,18 @@ type ListenerEntry struct {
 
 	// Port is the network port. Multiple listeners may use the
 	// same port, subject to the Listener compatibility rules.
-	Port PortNumber `json:"port"`
+	//
+	// If the port is not set or specified as zero, the implementation will assign
+	// a unique port. If the implementation does not support dynamic port
+	// assignment, it MUST set `Accepted` condition to `False` with the
+	// `UnsupportedPort` reason.
+	//
+	// +optional
+	//
+	// +kubebuilder:default=0
```

Comment:
If the default is now 0, is this not a backward compatible change once it goes to standard?  Formerly the port was optional, so there could be widespread omitted-port cases that were already properly handled.  Now there could be a completely different way that missing ports are handled.  Do we need to worry about that?

Reply:
> If the default is now 0, is this not a backward compatible change once it goes to standard?

I don't think so the ListenerSet is an alpha API so I believe we can make these changes. 

> Formerly the port was optional

It wasn't actually - maybe that's the confusion here